designs/2020-cache-contents-option/README.md

```diff
@@ -0,0 +1,58 @@
+- Start Date: 2020-07-15
+- RFC PR: https://github.com/eslint/rfcs/pull/63
+- Authors: @c-home
+
+# Allow use of file contents for cache
+
+## Summary
+
+This RFC proposes improving cache for CI by providing an option to use a hash (instead of `mtime` and `fsize`) comparison to determine whether a file has changed. 
+
+## Motivation
+
+Motivation for this change is the same as [#11490](https://github.com/eslint/eslint/issues/11490), where details of files like `mtime` are not preserved in the CI environment. With a hash comparison, the cache can be used in CI and would significantly reduce the time ESLint takes to run.
+
+ESLint uses [`file-entry-create`](https://github.com/royriojas/file-entry-cache) as its cache provider. [fileEntryCache#create](https://github.com/royriojas/file-entry-cache#createcachename-directory-usechecksum) gives the option to use a md5 hash through the `useChecksum` argument, but ESLint currently does not utilize it.
+
+## Detailed Design
+
+This RFC adds a `--cache-strategy` CLI option. Users can specify the option to be:
```

Comment:
Can you please describe the details of the implementation here? Which files would be changed?